lib/jpegstream.js

```diff
@@ -30,33 +31,43 @@ var Stream = require('stream').Stream;
  */
 
 var JPEGStream = module.exports = function JPEGStream(canvas, options, sync) {
-  var self = this
-    , method = sync
+  if (!(this instanceof JPEGStream)) {
+    throw new TypeError("Class constructors cannot be invoked without 'new'");
+  }
+
+  Readable.call(this);
+
+  var self = this;
+  var method = sync
       ? 'streamJPEGSync'
       : 'streamJPEG';
   this.options = options;
   this.sync = sync;
   this.canvas = canvas;
-  this.readable = true;
+
   // TODO: implement async
   if ('streamJPEG' == method) method = 'streamJPEGSync';
-  process.nextTick(function(){
-    canvas[method](options.bufsize, options.quality, options.progressive, function(err, chunk){
-      if (err) {
-        self.emit('error', err);
-        self.readable = false;
-      } else if (chunk) {
-        self.emit('data', chunk);
-      } else {
-        self.emit('end');
-        self.readable = false;
-      }
-    });
-  });
-};
 
-/**
- * Inherit from `EventEmitter`.
- */
+  // For now we're not controlling the c++ code's data emission, so we only
+  // call canvas.streamPNGSync once and let it emit data at will.
+  var hasStarted = false;
+
+  self._read = function () {
+    if (!hasStarted) {
+      hasStarted = true;
+      process.nextTick(function(){
+        canvas[method](options.bufsize, options.quality, options.progressive, function(err, chunk){
+          if (err) {
+            self.emit('error', err);
+          } else if (chunk) {
+            self.push(chunk);
+          } else {
+            self.push(null);
+          }
+        });
+      });
+    }
+  };
```

Comment:
allocating `read` here or in the prototype has few differences, apart from style. I prefer to use `prototype` when dealing with a class an inheriting.

Regarding this block, I would consider removing the `process.nextTick`  call. Plus, I think it's possible to recycle (declare inside the constructor) the function passed to `canvas[method]`.

The golden rule is "try not to allocate functions inside `_read`, `_write` or `_transform`". That is usually a very hot path and all those function allocation will slow you down.